bundles/org.eclipse.equinox.simpleconfigurator/src/org/eclipse/equinox/internal/simpleconfigurator/ConfigApplier.java

```diff
@@ -115,11 +120,91 @@ void install(URL url, boolean exclusiveMode) throws IOException {
 					if (additionalRefresh.length > 0)
 						refreshPackages(additionalRefresh, manipulatingContext);
 				}
+			}
+			// when refreshing large sets of bundles the resolver sometimes take a choice
+			// where a requirement can't be bound even though it is there and resolvable
+			// this code collects all those requirements and refresh the smaller set of
+			// bundles that provide these until a max retry is reached, all bundles are
+			// resolved, or all providers have been tried to refresh already
+			Set<Bundle> bundlesTried = new HashSet<>();
+			Collection<Bundle> provider;
+			Set<Bundle> doNotRefresh = getDoNotRefresh();
+			int maxtry = MAX_REFRESH_TRY;
```

Comment:
Can you explain why this would need to be tried more than once?  If it didn't resolve after the first attempt to refresh providers, why would it work on subsequent retries?